### DISCUSSION TITLE
Multiple AUTH

### DISCUSSION THREAD

**Discussion by keskinadem** [strongswan/strongswan#1697] [Q&A] [answered]
Does strongswan supports multiple authentication by multiple certificates? Sender signs AUTH payload with its private key. Can we double it with one more certificate and its private key in single IKE_AUTH exchange? (In this case sender should send 2 CERT payload)

**Comment by tobiasbrunner** [comment, marked as answer]
No, not in a single IKE_AUTH exchange. But strongSwan supports multiple authentication rounds per [RFC 4739](https://datatracker.ietf.org/doc/html/rfc4739).

**Reply by keskinadem** [reply]
RFC is experimental. As I understood each peer can use 2 certificates with two different signature algorithms. So, and it is like double IKE_AUTH exchange? And do you have test case or pcap output about it?

**Reply by tobiasbrunner** [reply]
Yes, there will be multiple IKE_AUTH exchanges, which is also the case when using EAP, by the way, the difference is that there are multiple AUTH payloads in these IKE_AUTH messages (with EAP there is only one at the end). There is the [ikev2/mult-auth-rsa-eap-sim-id](https://www.strongswan.org/testing/testresults/ikev2/mult-auth-rsa-eap-sim-id/) test scenario, where carol first authenticates with a certificate and then with EAP-SIM. There are also several example exchanges in the RFC.

**Reply by keskinadem** [reply]
In RFC there are 2 examples that only one peer use 2 different certificates.(initiator use 2 cert and responder 1 cert or vice verse: example 3 and 4).  But is this case possible? Initiator use 2 different certificates and responder uses different 2 certificates. As I see, it did not specified in RFC.

**Reply by tobiasbrunner** [reply]
Yes, both sides can use multiple authentication rounds (basically combining examples 3 and 4 in the RFC).

**Reply by keskinadem** [reply]
really appreciate for your answers, thank you.

**Comment by keskinadem** [comment]


**Reply by tobiasbrunner** [reply]
It's not standard-compliant. You obviously can hack something together like that, but it won't be interoperable.
